main.go

```diff
@@ -6,299 +6,34 @@
 package main
 
 import (
-	"bytes"
-	"fmt"
 	"log"
-	"math"
 	"os"
-	"regexp"
-	"sort"
 	"strconv"
-	"strings"
-	"time"
 
-	"github.com/google/gops/goprocess"
-	"github.com/shirou/gopsutil/v3/process"
-	"github.com/xlab/treeprint"
+	"github.com/google/gops/internal/cmd"
 )
 
-const helpText = `gops is a tool to list and diagnose Go processes.
-
-Usage:
-  gops <cmd> <pid|addr> ...
-  gops <pid> # displays process info
-  gops help  # displays this help message
-
-Commands:
-  stack      Prints the stack trace.
-  gc         Runs the garbage collector and blocks until successful.
-  setgc	     Sets the garbage collection target percentage.
-  memstats   Prints the allocation and garbage collection stats.
-  version    Prints the Go version used to build the program.
-  stats      Prints runtime stats.
-  trace      Runs the runtime tracer for 5 secs and launches "go tool trace".
-  pprof-heap Reads the heap profile and launches "go tool pprof".
-  pprof-cpu  Reads the CPU profile and launches "go tool pprof".
-
-All commands require the agent running on the Go process.
-"*" indicates the process is running the agent.`
-
-// TODO(jbd): add link that explains the use of agent.
-
 func main() {
-	if len(os.Args) < 2 {
-		processes()
-		return
-	}
-
-	cmd := os.Args[1]
-
-	// See if it is a PID.
-	pid, err := strconv.Atoi(cmd)
-	if err == nil {
-		var period time.Duration
-		if len(os.Args) >= 3 {
-			period, err = time.ParseDuration(os.Args[2])
-			if err != nil {
-				secs, _ := strconv.Atoi(os.Args[2])
-				period = time.Duration(secs) * time.Second
-			}
+	var root = cmd.NewRoot()
+	root.AddCommand(cmd.ProcessCommand())
+	root.AddCommand(cmd.TreeCommand())
+	root.AddCommand(cmd.AgentCommands()...)
+
+	// Legacy support for `gops <pid>` command.
+	//
+	// When the second argument is provided as int as opposed to a sub-command
+	// (like proc, version, etc), gops command effectively shortcuts that
+	// to `gops process <pid>`.
+	if len(os.Args) > 1 {
```

Comment:
I didn't want to break existing `gops <pid>`, and this is the most straight forward way I could implement it. Cobra doesn't support anything like this, so I had to be done manually